src/js/actions/layers.js

```diff
@@ -1694,10 +1684,13 @@ define(function (require, exports) {
 
         var numberRef = (typeof target === "number"),
             targetRef = numberRef ? layerLib.referenceBy.index(target) : layerLib.referenceBy[target],
-            reorderObj = layerLib.reorder(layerRef, targetRef),
-            reorderPromise = descriptor.playObject(reorderObj);
+            reorderObj = layerLib.reorder(layerRef, targetRef);
       
-        return reorderPromise.bind(this)
+        return this.transfer(historyActions.newHistoryState, document.id, strings.ACTIONS.REORDER_LAYERS)
```

Comment:
The problem here is that we ARE still getting historyState events from PS even if we are initiating the change.  I like to prevent, when possible, a race condition in how the history state gets incremented.  By waiting for this new history state to be created in our model *before* playing the action in PS, we insure that the resultant historyState event from PS will be a no-op.  Maybe we should talk about this in more detail, but that's my story for now.

I could be persuaded that it is OK to use the "amend rogue" flow even for these optimistic style actions.